blockchain/blockchain.go

```diff
@@ -608,27 +607,29 @@ func (bc *blockchain) startExistingBlockchain(ctx context.Context) error {
 	}
 	log.L().Info("Restarting blockchain.",
 		zap.Uint64("chainHeight",
-			bc.tipHeight),
+			tipHeight),
 		zap.Uint64("factoryHeight", stateHeight))
 	return nil
 }
 
 func (bc *blockchain) tipInfo() (*protocol.TipInfo, error) {
-	if bc.tipHeight == 0 {
+	tipHeight, _ := bc.dao.GetTipHeight()
+	if tipHeight == 0 {
 		return &protocol.TipInfo{
 			Height:    0,
 			Hash:      bc.config.Genesis.Hash(),
 			Timestamp: time.Unix(bc.config.Genesis.Timestamp, 0),
 		}, nil
 	}
-	header, err := bc.dao.Header(bc.tipHash)
+	tipHash, _ := bc.dao.GetTipHash()
+	header, err := bc.dao.Header(tipHash)
 	if err != nil {
 		return nil, err
 	}
```

Comment:
to handle this case, two ways:
1. dao.TipHeight(), and then use the tip height to read header
2. dao.TipHeader(), and then use the tip header to get tip height (header->Height()) and hash (header->HashBlock())